src/RMPI/helpers.jl

```diff
@@ -79,11 +79,13 @@ function Rimu.sort_into_targets!(dtarget::MPIData, source::AbstractDVec)
     ltarget = localpart(dtarget)
     empty!(ltarget) # start with empty slate
     strategy = dtarget.s
-    P = pairtype(source) # compute pairtype
-    # println("pairtype = ",P)
-    sort_into_targets!(ltarget, source, P, strategy)
+    sort_into_targets!(ltarget, source, strategy)
 end
 
+# This function is just a wrapper that makes allreduce treat a SVector as a scalar
+function _communicate_stats(stats, comm)
+    return invoke(MPI.Allreduce, Tuple{Any, typeof(+), MPI.Comm}, stats, +, comm)
+end
```

Comment:
This looks very clever. Is this safe, though? It looks like we are relying on an implementation detail of `MPI.Allreduce`, which could change in the future.

Reply:
This is needed because `MPI.Allreduce` is overloaded for `AbstractArrays`, but we want to treat `stats` as a scalar. It essentially tells `MPI.Allreduce` to treat `stats` as a value of a type `MPI` does not know about (as if it were e.g. some kind of user-defined type). It's safe as long as the signature of `MPI.Allreduce` doesn't change.
I agree it's a bit clever, but the alternative approach I tried with `StaticArrays.Scalar` didn't work for some reason.